core/frontend/src/tile/ThreeDTileFormatInterpreter.ts

```diff
@@ -0,0 +1,142 @@
+
+      if (worldToEcefTransform.matrix.isIdentity && (earthCenterToRangeCenterRayLenght < 1.0E5 || isNotNearEarthSurface)) {
+        isGeolocated = false;
+        worldRange.extendRange(Range3d.fromJSON(ecefRange));
+        const centerOfEarth: EcefLocationProps = { origin: { x: 0.0, y: 0.0, z: 0.0 }, orientation: { yaw: 0.0, pitch: 0.0, roll: 0.0 } };
+        location = centerOfEarth;
+        Logger.logTrace(loggerCategory, "RealityData NOT Geolocated", () => ({ ...location }));
+      } else {
+        let ecefLocation: EcefLocation;
+        const locationOrientation = YawPitchRollAngles.tryFromTransform(worldToEcefTransform);
+        // Fix Bug 445630: [RDV][Regression] Orientation of georeferenced Reality Mesh is wrong.
+        // Use json.root.transform only if defined and not identity -> otherwise will use a transform computed from cartographic center.
+        if (!worldToEcefTransform.matrix.isIdentity && locationOrientation !== undefined && locationOrientation.angles !== undefined)
+          ecefLocation = new EcefLocation({ origin: locationOrientation.origin, orientation: locationOrientation.angles.toJSON() });
+        else
+          ecefLocation = EcefLocation.createFromCartographicOrigin(cartoCenter!);
+        location = ecefLocation;
+        Logger.logTrace(loggerCategory, "RealityData is worldToEcefTransform.matrix.isIdentity", () => ({ isIdentity: worldToEcefTransform!.matrix.isIdentity }));
+        // iModelDb.setEcefLocation(ecefLocation);
+        const ecefToWorld = ecefLocation.getTransform().inverse()!;
+        worldRange.extendRange(Range3d.fromJSON(ecefToWorld.multiplyRange(ecefRange)));
+        Logger.logTrace(loggerCategory, "RealityData ecefToWorld", () => ({ ...ecefToWorld }));
+      }
+    }
+    const spatialLocation: SpatialLocationAndExtents = { location, worldRange, isGeolocated };
+    return spatialLocation;
+  }
+  /** Gets information to identify the product and engine that create this reality data
+   * Will return undefined if cannot be resolved
+   * @param rootDocjson root document file in json format
+   * @returns information to identify the product and engine that create this reality data
+   * @alpha
+   */
+  public static getPublisherProductInfo(rootDocjson: any): PublisherProductInfo {
+    const info: PublisherProductInfo = {product: "", engine: "", version: "", rootChildren: ""};
+    if (rootDocjson && rootDocjson.root) {
+      info.rootChildren = rootDocjson.root.children.length;
+      if (rootDocjson.root.SMPublisherInfo) {
+        info.product = rootDocjson.root.SMPublisherInfo.Product ? rootDocjson.root.SMPublisherInfo.Product : "";
+        info.engine =  rootDocjson.root.SMPublisherInfo.Publisher ? rootDocjson.root.SMPublisherInfo.Publisher : "";
+        info.version = rootDocjson.root.SMPublisherInfo["Publisher Version"] ? rootDocjson.root.SMPublisherInfo["Publisher Version"] : "" ;
+      }
+    }
+    return info;
+  }
+}
+
+class RealityModelTileUtils {
```

Comment:
nitpick but this namespace serves no purpose.

Reply:
Remove namespace RealityModelTileUtils and add method to ThreeDTileFormatInterpreter class instead.